url594c007908_hugobraga_College_tgz-pJ8-main_GraphJ8/src/dataStructure/AdjList.java

```diff
@@ -0,0 +1,352 @@
+    		ALNode node = adjl[index];
+    		if (node == null)
+    			continue;
+    		System.out.print(index+": [");
+    		while (node != null) {
+    			levels.add(node.getVid());
+    			System.out.print(node.getVid()+", ");
+    			node = node.getProx();
+    		}
+    		System.out.println("]");
+    	}
+    }
+
+    public void printAdjList() {
+	int i;
+	ALNode node;
+
+	System.out.println ("printing Adjacent List:");
+	for (i=0; i < nv; i++) {
+		if (adjl[i] == null)
+			continue;
+	    System.out.print ("Vertice: " + i + ": ");
+	    node = adjl[i];
+	    while (node != null) {
+	    	System.out.print(node.getVid()+"/c: "+node.getWeight()+", ");
+		node = node.getProx();
+	    }
+	    System.out.println();
+	}
+    }
+
+    public void setWeight(Weight w) {
+	ALNode node;
+
+	for (int i=0; i < nv; i++) {
+	    node = adjl[i];
+	    while (node != null) {
+		node.setWeight(w.getWeight(i, node.getVid(), this));
+		node = node.getProx();
+	    }
+	}
+    }
+
+    public  void checkCopyList (AdjList nadjl, int index) {
+    	ALNode p;
+
+    	for (ALNode q = nadjl.getList(index); q != null; q = q.getProx()) {
+    		ALNode temp = adjl[index];
+    		boolean flag = false;
+    		while (temp != null) {
+    			if (temp.getVid() == q.getVid()) {
+    				flag = true;
+    				break;
+    			}
+    			temp = temp.getProx();
+    		}
+    		if (flag)
+    			continue;
+    	    p = new ALNode(q.getVid(), q.getWeight());
+    	    p.setProx(adjl[index]);
+    	    adjl[index] = p;
+    	}
+    }
+
+    public  void copyList (AdjList nadjl, int index) {
+	ALNode p;
+
+	for (ALNode q = nadjl.getList(index); q != null; q = q.getProx()) {
+	    p = new ALNode(q.getVid(), q.getWeight());
+	    p.setProx(adjl[index]);
+	    adjl[index] = p;
+	}
+    }
+
+
+    public void readFromTopologyFile (String fname, PositionTable pt) {
+	String s1, s2;
+
+	try {
+	    BufferedReader in = new BufferedReader(new FileReader(fname));
+	    String aLine1, aLine2;
+
+	    for (int i = 0; i < nv; i++) {
+		aLine1 = in.readLine();
+		aLine2 = in.readLine();
+		StringTokenizer stx1 = new StringTokenizer(aLine1);
+		StringTokenizer stx2 = new StringTokenizer(aLine2);
+
+		s1 = stx1.nextToken();    // read word 'Node:'
+		s1 = stx1.nextToken();    // read node id
+		s1 = stx1.nextToken();    // read word 'neighbours:'
+
+		s2 = stx2.nextToken();    // read word 'Node:'
+		s2 = stx2.nextToken();    // read node id
+		s2 = stx2.nextToken();    // read word 'edgecosts:'
+
+		while (stx1.hasMoreTokens()) {
+		    s1 = stx1.nextToken();    // read neighbour
+		    int nb = Integer.parseInt(s1);
+
+		    s2 = stx2.nextToken();    // read neighbour
+		    double nbc = Double.parseDouble(s2);
+
+		    insertEdge (i, nb, nbc);
+		}
+
+		aLine1 = in.readLine();   // skip line with optimized neighbour set
+	    }
+	    in.close();
+	} catch (IOException e) {
+	}
+    }
+	
+
+    // Methods related to the implementation of Prim's MST algorithm
+
+    public void MSTPrim() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.